src/tests/e2e/tron-test-tx-middleware.ts

```diff
@@ -0,0 +1,193 @@
+      type: 'function'
+    },
+    {
+      inputs: [],
+      payable: false,
+      stateMutability: 'nonpayable',
+      type: 'constructor'
+    },
+    {
+      anonymous: false,
+      inputs: [
+        {
+          indexed: true,
+          name: '_value',
+          type: 'uint256'
+        },
+        {
+          indexed: false,
+          name: 'sender',
+          type: 'address'
+        }
+      ],
+      name: 'NewValueSet',
+      type: 'event'
+    }
+  ]
+  
+    // Deploy the contract using loom provider
+    const result = await deployContract(loomProvider, contractData)
+
+    // Instantiate Contract using web3
+    const web3 = new Web3(loomProvider)
+    const contract = new web3.eth.Contract(ABI, result.contractAddress, {
+      from: LocalAddress.fromPublicKey(pubKey).toString()
+    })
+
+    // And get the tron signer
+    const fullNode = new TronWeb.providers.HttpProvider('http://127.0.0.1:9090');
+    const solidityNode = new TronWeb.providers.HttpProvider('http://127.0.0.1:9090');
+    const eventServer = 'http://127.0.0.1:9090';
+    const privateKey = '11b5b4c7ff9e69e89e66ef614d8d63a81f2799c3b2bad938a6ae3b0a10a12772'
+    const tronWeb = new TronWeb(fullNode, solidityNode, eventServer, privateKey);
+    const tronDefaultAddr = tronWeb.defaultAddress.base58
+    let trxAddress = tronWeb.address.toHex(tronDefaultAddr)
+    let trxAddressHex = '0x'+ trxAddress.substring(2,100)
+
+    const signer = new TronWebSigner(tronWeb, trxAddressHex)
+  
+    return { client, pubKey, privKey, signer, loomProvider, contract, ABI }
+  }
+
+test('Test Signed Trx Tx Middleware (Type 3)', async t => {
+  try {
+    const { client, signer, pubKey, loomProvider, contract } = await bootstrapTest(
+      createTestHttpClient
+    )
+
+    const addressMapper = await AddressMapper.createAsync(
+      client,
+      new Address(client.chainId, LocalAddress.fromPublicKey(pubKey))
+    )
+
+    // Set the mapping
+    const trxAddress = await signer.getAddress()
+    const from = new Address(client.chainId, LocalAddress.fromPublicKey(pubKey))
+    const to = new Address('tron', LocalAddress.fromHexString(trxAddress))
+
+    // console.log("signer is ", signer)
+    // Add mapping if not added yet
+    if (!(await addressMapper.hasMappingAsync(from))) {
+      await addressMapper.addIdentityMappingAsync(from, to, signer)
+    }
+
+    try {
+      const addressMapped = await addressMapper.getMappingAsync(from)
+      t.assert(addressMapped.from.equals(from), 'Should be mapped the from address')
+      t.assert(addressMapped.to.equals(to), 'Should be mapped the to address')
+    } catch (err) {
+      t.error(err)
+    }
+
+    const callerChainId = 'tron'
+    // Override the default caller chain ID
+    loomProvider.callerChainId = callerChainId
+    // Ethereum account needs its own middleware
+    loomProvider.setMiddlewaresForAddress(to.local.toString(), [
+      new CachedNonceTxMiddleware(pubKey, client), // FIX
```

Comment:
Fix what?

Reply:
I can remove this but should I be worried that it's also in `client-test-tx-middleware.ts` too?